src/twisted/internet/task.py

```diff
@@ -780,24 +781,16 @@ def seconds(self):
         return self.rightNow
 
 
-    def _sortCalls(self):
-        """
-        Sort the pending calls according to the time they are scheduled.
-        """
-        self.calls.sort(key=lambda a: a.getTime())
-
-
     def callLater(self, when, what, *a, **kw):
         """
         See L{twisted.internet.interfaces.IReactorTime.callLater}.
         """
         dc = base.DelayedCall(self.seconds() + when,
-                               what, a, kw,
-                               self.calls.remove,
-                               lambda c: None,
-                               self.seconds)
-        self.calls.append(dc)
-        self._sortCalls()
+                              what, a, kw,
+                              lambda c: None,
+                              lambda c: None,
+                              self.seconds)
+        heapq.heappush(self.calls, (dc.getTime(), dc))
```

Comment:
`Clock.calls` is part of `Clock`'s public API, so changing its internal structure breaks calling code including Twisted's own tests:
https://travis-ci.org/twisted/twisted/jobs/488068561#L4517

Fortunately, [reactors have used heaps to sort `DelayedCall`s](https://github.com/twisted/twisted/blob/411d08f688bd8cdf61650e0ea1b24e99e14330e0/src/twisted/internet/base.py#L823-L830) for [a long time](https://twistedmatrix.com/trac/ticket/707), so the class implements the [the methods necessary for ordering](https://github.com/twisted/twisted/blob/411d08f688bd8cdf61650e0ea1b24e99e14330e0/src/twisted/internet/base.py#L164-L181).

That means you can push the `DelayedCall` onto the heap directly:

```suggestion
        heapq.heappush(self.calls, dc)
```